glue/utils/qt/autocomplete_widget.py

```diff
@@ -0,0 +1,127 @@
+# Code adapted from:
+#
+# http://rowinggolfer.blogspot.de/2010/08/qtextedit-with-autocompletion-using.html
+#
+# and based on:
+#
+# http://qt-project.org/doc/qt-4.8/tools-customcompleter.html
+
+from ...external.qt import QtGui, QtCore
+
+
+__all__ = ["CompletionTextEdit"]
+
+try:  # Python 2
+
+    QString = QtCore.QString
```

Comment:
So we're using version 2 of the Qt api, which is supposed to do away with QString and just use unicode instead. Is this really needed here? I suspect there's a cleaner way that avoids QString